internal/app/osd/internal/reg/reg.go

```diff
@@ -41,7 +41,8 @@ import (
 // Registrator is the concrete type that implements the factory.Registrator and
 // proto.OSDServer interfaces.
 type Registrator struct {
-	Data *userdata.UserData
+	Data       *userdata.UserData
+	InitClient *InitServiceClient
```

Comment:
nit: I think I'm a fan of embedding this so we can just do `r.Reboot` or `r.Shutdown`.

Reply:
I think this won't work as they have actually different signatures (even if I embed directly `init` API), as they're coming from different `proto` packages.

Reply:
I thought if we could have simply said that `OSD` API is a superset of `Init` API, I think it should work with protobufs? But implementation in OSD would be different anyways, as it should proxy to init.

Reply:
looks like no service inheritance :(

https://stackoverflow.com/questions/39196368/grpc-service-inheritance

Reply:
ended up doing this embedding and plus making OSD implement Init API